postgresql/database.go

```diff
@@ -84,79 +78,242 @@ func debugLog(query string, args []interface{}, err error, start int64, end int6
 	}
 }
 
-func init() {
+// Returns the underlying *sqlx.DB instance.
+func (s *source) Driver() interface{} {
```

Comment:
Some pieces of user code expect this interface to be a `*sql.DB`, and the pattern they use is something like this:

```
  drv = sess.Driver().(*sql.DB)
```

Changing it to *sqlx.DB could cause lots of noise (nil pointer method calls), as the cast to `*sql.DB` would now be `nil`.

Right now our main test fails because it still expects the value of `Driver()` to be a `*sql.DB`, we can change it easily, but our users may get a nasty surprise.

I think we have some options:

a) Find a way in which both `drv.(*sqlx.DB)` and `drv.(*sql.DB)` work (this would be awesome).

or 

b) Stick to `Driver()` being `*sql.DB`, users will always be able to generate a `*sqlx.DB` with [sqlx.NewDb](http://godoc.org/github.com/jmoiron/sqlx#NewDb)

I'm looking to get `a)` woking, I'm trying to put `sqlx.DB` and `sql.DB` as anonymous fields of the same struct, but the compiler refuses to build anything because the types have the same name. I'll keep trying.